Article 116: Esoteric Programming Languages

ImarinArwen opened the pull request:
## Attributes

<!-- Please use one of these items per category by removing the comment tag at the beginning and end of the line. -->

#### Length of text ####
 :page_with_curl: short 
<!-- :book:  long -->

####  Difficulty to read ####
 :mouse:  light 
<!-- :elephant:  heavy -->

####  Difficulty to understand #### 
 :school_satchel:  easy 
<!-- :mortar_board:  difficult -->

####  Type of text #### 
<!-- :laughing:  funny -->
 :bowtie:  informative 
<!-- :sunglasses:  catchy -->

####  Type of header #### 
<!-- :santa:  funny --> 
<!-- :bulb:  informative -->
 :boom:  catchy

Fjolyo commented:
# Plenum Suggestions

- Last sentence is confusing (... write *any* program ...)
- Maybe don't mention "Software ENgineering"
- Maybe mention the limit of programming languages
- Express how many comments are needed for the code

ImarinArwen commented:
Added the any in the last sentence.
Changed the SE part. Should I explain the unusual language concepts?
Added one sentence about the number of comments needed. 

I did not mention the limit of programming languages. Because I'm not sure how to do this in just one sentence.

MoiraZuber commented:
Added the new image. No significant changes.

marcelluethi commented on articles/article-116-esoteric-programming-languages.md:
```diff
@@ -0,0 +1,27 @@
+
+<  > + -  . , [ ]
+
+There are more comments needed to explain the code than the size of the code itself.
```
Suggestion: Maybe we could write: To be able to understand the code, more comments are needed to explain the code than the size of the code itself.

ImarinArwen replied:
done.

marcelluethi commented on articles/article-116-esoteric-programming-languages.md:
```diff
@@ -0,0 +1,27 @@
+<  > + -  . , [ ]
+
+There are more comments needed to explain the code than the size of the code itself.
+It is possible to write any program with this language the question solely is if one wants to.
```
Language suggestion: It is possible to write any program with this language. But the question is, if one really wants.

ImarinArwen commented:
I changed the last two sentences as suggested

marcelluethi commented:
@ImarinArwen great, thanks. I guess it's ready for testing.

noverney commented on articles/article-116-esoteric-programming-languages.md:
```diff
@@ -0,0 +1,33 @@
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+There are thousands of programming languages. But some of them are just laborious and inefficient. They are not designed for practical usage but to implement different aspects of unusual language concepts. Those programming languages are called esoteric. One of them is called Brainfuck. It consists of the eight language commands:
```
I would expand a bit on the first two sentences:
Since the advent of computers, thousands of programming languages have been invented. While some are great and are still used today, other languages are laborious and inefficient.

noverney replied:
might want to change the word esoteric to  esoteric programming language (sometimes shortened to esolang)

ImarinArwen replied:
Changed the first two sentences

noverney commented on articles/article-116-esoteric-programming-languages.md:
```diff
@@ -0,0 +1,33 @@
+<  > + -  . , [ ]
+
+To be able to understand the code, more comments are needed to explain the code than the size of the code itself.
+It is possible to write any program with this language. But the question is, if one really wants.
```
But the question is, if really wants to.

ImarinArwen replied:
Changed this part

noverney replied:
My mistake my comment should have been:
But the question is, if one really wants to.
